lock-impl/src/main/java/com/palantir/lock/impl/LockServiceImpl.java

```diff
@@ -233,26 +235,48 @@ public static LockServiceImpl create() {
     public static LockServiceImpl create(LockServerOptions options) {
         com.palantir.logsafe.Preconditions.checkNotNull(options);
         ExecutorService newExecutor = PTExecutors.newCachedThreadPool(LockServiceImpl.class.getName());
-        return create(options, Ownable.owned(newExecutor));
+        return create(
+                options, Ownable.owned(newExecutor), Refreshable.only(LockServerConfigs.DEFAULT_THREAD_INFO_CONFIG));
     }
 
     public static LockServiceImpl create(LockServerOptions options, ExecutorService injectedExecutor) {
         com.palantir.logsafe.Preconditions.checkNotNull(options);
-        return create(options, Ownable.notOwned(injectedExecutor));
+        return create(
+                options,
+                Ownable.notOwned(injectedExecutor),
+                Refreshable.only(LockServerConfigs.DEFAULT_THREAD_INFO_CONFIG));
     }
 
-    private static LockServiceImpl create(LockServerOptions options, Ownable<ExecutorService> executor) {
+    public static LockServiceImpl create(
+            LockServerOptions options,
+            ExecutorService injectedExecutor,
+            Refreshable<DebugThreadInfoConfiguration> threadInfoConfiguration) {
+        com.palantir.logsafe.Preconditions.checkNotNull(options);
+        return create(options, Ownable.notOwned(injectedExecutor), threadInfoConfiguration);
+    }
+
+    private static LockServiceImpl create(
+            LockServerOptions options,
+            Ownable<ExecutorService> executor,
+            Refreshable<DebugThreadInfoConfiguration> threadInfoConfiguration) {
         if (log.isTraceEnabled()) {
             log.trace("Creating LockService with options={}", SafeArg.of("options", options));
         }
         final String jmxBeanRegistrationName = "com.palantir.lock:type=LockServer_" + instanceCount.getAndIncrement();
         LockServiceImpl lockService = new LockServiceImpl(
-                options, () -> JMXUtils.unregisterMBeanCatchAndLogExceptions(jmxBeanRegistrationName), executor);
+                options,
+                () -> JMXUtils.unregisterMBeanCatchAndLogExceptions(jmxBeanRegistrationName),
+                executor,
+                threadInfoConfiguration);
         JMXUtils.registerMBeanCatchAndLogExceptions(lockService, jmxBeanRegistrationName);
         return lockService;
     }
 
-    private LockServiceImpl(LockServerOptions options, Runnable callOnClose, Ownable<ExecutorService> executor) {
+    private LockServiceImpl(
+            LockServerOptions options,
+            Runnable callOnClose,
+            Ownable<ExecutorService> executor,
+            Refreshable<DebugThreadInfoConfiguration> threadInfoConfiguration) {
```

Comment:
I would propose either renaming this to be `ReloadableLockServerOptions`, which would contain `DebugThreadInfoConfiguration`. Or, just making `LockServerOptions` refreshable, with the understanding that only certain properties are actually refreshable themselves.

Reply:
Went with the later option to avoid creating a new class.